### PR TITLE
Add Transact-SQL dialect (WIP)

### PR DESCRIPTION
**Pull request by GeorgeSittas**
This is an attempt to add support for the transact sql dialect. Feedback is welcome. Plan to add more features soon.

### CODE REVIEW

File: sqlglot/dialects/dialect.py [hunk 1]
```diff
@@ -26,6 +26,7 @@ class Dialects(str, Enum):
     SQLITE = "sqlite"
     STARROCKS = "starrocks"
     TABLEAU = "tableau"
+    TRANSACTSQL = "transactsql"
```
Comment (tobymao) [inline sqlglot/dialects/dialect.py:29]: maybe let's just call it TSQL ?

File: sqlglot/tokens.py [hunk 2]
```diff
@@ -123,6 +130,7 @@ class TokenType(AutoName):
     CURRENT_TIME = auto()
     CURRENT_TIMESTAMP = auto()
     DIV = auto()
+    DECLARE = auto()
```
Comment (tobymao) [inline sqlglot/tokens.py:133]: why did you add this it's not used?

Comment (GeorgeSittas) [reply sqlglot/tokens.py:133]: Yeah I'll remove it, I was trying out the declare syntax and forgot it. We can add it again later if needed.

File: sqlglot/dialects/dialect.py (hunk 1 above)
Comment (GeorgeSittas) [reply sqlglot/dialects/dialect.py]: sure

**Review by tobymao — APPROVED** [review]
